model/metadata/service.go

```diff
@@ -67,35 +69,36 @@ type node struct {
 }
 
 //DecodeService decodes a given input into a Service instance
-func DecodeService(input interface{}, err error) (*Service, error) {
+func DecodeService(input interface{}, hasShortFieldNames bool, err error) (*Service, error) {
 	if input == nil || err != nil {
 		return nil, err
 	}
 	raw, ok := input.(map[string]interface{})
 	if !ok {
 		return nil, errors.New("invalid type for service")
 	}
+	field := fields.Mapper(hasShortFieldNames)
```

Comment:
You could initialize the decoder with the mapping information `decoder(fields.Mapper(hasShortFieldNames))`, allowing to abstract away the mapper inside the decoding.

Reply:
I tried that, but there are a lot of small functions (`decodeHTTP`, `decodeService`, `decodeStacktrace`, etc) that would need to be adapted to be methods instead, meaning that they would not be reusable (eg. duplicated `metadata.DecodeService`,  `transaction.DecodeService`, `span.DecodeService`, etc.). 
At the end thought it was not worth it and this produced the smallest diff.